### PR TITLE
OCM-6646 | fix: detect presence of --help processing args

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by hunterkepley** [comment]
LGTM, 
only concern is if it will print the help for the subcommands (such as `rosa list --help` vs `rosa list clusters --help`

**Comment by ciaranRoche** [comment]
@thomasmckay can you ensure the ticket meets the JIRA definition of ready - https://docs.google.com/document/d/1UxSahjnmt4Ovj9y2hxcJt082j8KvJLC7CTfLKSs4HMA/edit#bookmark=id.rgd3xpx7hk0c

File: pkg/arguments/arguments.go [hunk 1]
```diff
@@ -195,21 +195,13 @@ func PreprocessUnknownFlagsWithId(cmd *cobra.Command, argv []string) error {
...
-	if helpVal {
-		return pflag.ErrHelp
-	}
-
 	foundId := false
 	for i, arg := range argv {
 		switch {
+		// If help is called, regardless of other flags, return we want help.
+		// Also say we need help if the command isn't runnable.
+		case arg == "--help":
```
Comment (robpblake) [inline pkg/arguments/arguments.go:203]: @thomasmckay Could you please update `arguments_test.go` with a unit test for this.

cheers,
Rob

**Comment by thomasmckay** [comment]
> LGTM, only concern is if it will print the help for the subcommands (such as `rosa list --help` vs `rosa list clusters --help`

it does work for this case

File: pkg/arguments/arguments.go (hunk 1 above)
Comment (thomasmckay) [reply pkg/arguments/arguments.go:203]: updated tests, thanks!

**Comment by thomasmckay** [comment]
> @thomasmckay can you ensure the ticket meets the JIRA definition of ready - https://docs.google.com/document/d/1UxSahjnmt4Ovj9y2hxcJt082j8KvJLC7CTfLKSs4HMA/edit#bookmark=id.rgd3xpx7hk0c

added priority and story points to jira

File: pkg/arguments/arguments.go (hunk 1 above)
Comment (robpblake) [reply pkg/arguments/arguments.go:203]: @thomasmckay I don't see the update tests file in the commits for this PR?

**Comment by ciaranRoche** [comment]
@thomasmckay just bumping this, can you commit the tests so we can merge this :pray:

**Comment by ciaranRoche** [comment]
@thomasmckay another bump on this one :pray:

**Pull request closed without merging**